template/cordova/lib/log.js

```diff
@@ -0,0 +1,226 @@
+/*
+       Licensed to the Apache Software Foundation (ASF) under one
+       or more contributor license agreements.  See the NOTICE file
+       distributed with this work for additional information
+       regarding copyright ownership.  The ASF licenses this file
+       to you under the Apache License, Version 2.0 (the
+       "License"); you may not use this file except in compliance
+       with the License.  You may obtain a copy of the License at
+
+         http://www.apache.org/licenses/LICENSE-2.0
+
+       Unless required by applicable law or agreed to in writing,
+       software distributed under the License is distributed on an
+       "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+       KIND, either express or implied.  See the License for the
+       specific language governing permissions and limitations
+       under the License.
+*/
+
+var path    = require('path'),
+    et      = require('elementtree'),
+    Q       = require('q'),
+    cp      = require('child_process'),
+    shelljs = require('shelljs');
+
+var appTracingLogInitialState = null,
+    adminLogInitialState = null;
+
+/*
+ * Gets windows AppHost/ApplicationTracing and AppHost/Admin logs
+ * and prints them to console
+ */
+module.exports.run = function() {
+    getLogState('Microsoft-Windows-AppHost/ApplicationTracing').then(function (state) {
+        appTracingLogInitialState = state;
+        return getLogState('Microsoft-Windows-AppHost/Admin');
+    }).then(function(state) {
+        adminLogInitialState = state;
+        return enableLogging('Microsoft-Windows-AppHost/Admin');
+    }).then(function () {
+        return enableLogging('Microsoft-Windows-AppHost/ApplicationTracing');
+    }).then(function () {
+        console.log('Logging enabled, now printing logs. To stop, please press Ctrl+C once.');
+        startLogging('Microsoft-Windows-AppHost/Admin');
+        startLogging('Microsoft-Windows-AppHost/ApplicationTracing');
+    }).catch(function (error) {
+        console.error(error);
+    });
+
+    // Disable logs before exiting
+    process.on('exit', function () {
+        if ((appTracingLogInitialState !== null) && !appTracingLogInitialState) {
+            disableLogging('Microsoft-Windows-AppHost/ApplicationTracing');
+        }
+        if ((adminLogInitialState !== null) && !adminLogInitialState) {
+            disableLogging('Microsoft-Windows-AppHost/Admin');
+        }
+    });
+
+    // Catch Ctrl+C event and exit gracefully
+    process.on('SIGINT', function () {
+        process.exit(2);
+    });
+
+    // Catch uncaught exceptions, print trace, then exit gracefully
+    process.on('uncaughtException', function(e) {
+        console.log(e.stack);
+        process.exit(99);
+    });
+};
+
+function startLogging(channel) {
+    var startTime = new Date().toISOString();
+    setInterval(function() {
+        var command = 'wevtutil qe ' + channel + ' /q:"*[System [(TimeCreated [@SystemTime>\'' + startTime + '\'])]]" /e:root';
```

Comment:
@robpaveza Actually, I just modified the script so that it wouldn't fail if it hasn't administrator privileges, but only warn about that and try to show the logs that are already enabled.